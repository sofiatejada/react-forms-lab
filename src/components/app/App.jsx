import React from 'react';
import NewsSearch from '../../containers/NewsSearch';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

export default function App() {
  return (
    <BrowserRouter>
      <div>
        <Switch>
          <Route exact path="/" component={NewsSearch} />
          {/* <Route exact path="/:title" component={NewsSearch} /> */}
        </Switch>
      </div>
    </BrowserRouter>
  )
}
