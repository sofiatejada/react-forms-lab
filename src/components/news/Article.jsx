import React from 'react';
import PropTypes from 'prop-types';

const Article = ({ title, author, description }) => (
  <div>
    <h3>{title}</h3>
    <p>{author}</p>
    <p>{description}</p>
  </div>
);

Article.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default Article;
