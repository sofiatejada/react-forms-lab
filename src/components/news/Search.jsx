import React from "react";
import PropTypes from 'prop-types';

const Search = ({ search, onSearchChange }) => (
  <div>
    <input type="text" value={search} onChange={onSearchChange} aria-label="search" />
  </div>
);

Search.propTypes = {
  search: PropTypes.string.isRequired,
  onSearchChange: PropTypes.func.isRequired,
};
