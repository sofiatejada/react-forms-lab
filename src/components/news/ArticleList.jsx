import React from 'react';
import { PropTypes } from 'prop-types';
import Article from './Article';

const ArticleList = ({ articles }) => (
  <ul aria-label="articles">
    {articles.map((article, index) => (
      <li key={`${article}-${index}`}>
        <Article title={article.title} author={article.author} description={article.description} />
      </li>
    ))}
  </ul>
);

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  )
}

export default ArticleList;
