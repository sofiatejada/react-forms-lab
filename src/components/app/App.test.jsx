import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import App from './App';
import Article from '../news/Article';
import ArticleList from '../news/ArticleList';
import Search from '../news/Search';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Article component', () => {
  it('displays an article', () => {
    render(
      <Article 
        title="Look at all those chickens" 
        author="Pikachu" 
        description="It's actually a flock of geese..." />
    );

    const article = screen.getByText('It\'s actually a flock of geese...');
    expect(article).toMatchSnapshot();
  });
});

describe('ArticleList component', () => {
  it('displays a list of articles', () => {
    render(
      <ArticleList
        articles={[
          {
            title: 'I can\'t swim',
            author: 'Raven',
            description: 'It\'s her birthday but she can\'t swim'
          },
          {
            title: 'What did you do?',
            author: 'Grandma',
            description: 'I do nothing wrong in the playroom :['
          }
        ]} 
      />
    );

    const ul = screen.getByRole('list', { name: 'articles' });
    expect(ul).toMatchSnapshot();
  });
});

describe('Search', () => {
  it('displays form with one input and a button', () => {
    render(
      <Search 
        search="Memes"
        onSearchChange="{() => {}}"
        onSearchSubmit="{() => {}}"
      />
    );

    const searchForm = screen.getByTestId('search-form');
    expect(searchForm).toMatchSnapshot();
  });
});
