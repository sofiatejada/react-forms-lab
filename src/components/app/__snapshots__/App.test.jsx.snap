// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <form>
          <input
            aria-label="search"
            type="text"
            value="pokemon"
          />
          <button
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
      <ul
        aria-label="articles"
      />
    </div>
  </div>
</DocumentFragment>
`;
